skgstat/binning.py

```diff
@@ -15,17 +15,18 @@ def binify_even_width(X, N=10, w=None, dm=None, maxlag=None, **kwargs):
 
     :param X: 1D array of x, y coordinates.
     :param N: int with the number of bins
-    :param w: width of the bins
+    :param w: integer or float with width of the bins
     :param dm: numpy.ndarray with the distance matrix
     :param maxlag: maximum lag for the binning
     :param kwargs: will be passed to calculate the point_matrix if no dm is given
     :return: distance matrix
     """
 
     _X = list(X)
-    # check that all elements in the index have exactly a x and y coordinate
-    if any([not len(e) == 2 for e in _X]):
-        raise ValueError('The passed point data does not have a x and y coordinate for each point.')
+    # check that all elements in the index have exactly a x and y coordinate or are n dimensional
+
+    if not len(set([len(e) for e in _X])) == 1:
```

Comment:
and the length of (any) element has to be larger than 0 (or 1, if the distance functions can handle 1D coordinates)